### PR TITLE
[MINOR] Added leaky relu layer + grad check and updated 2NN example

### PR DESCRIPTION
**Pull request by tobiasrieger**
Added leaky relu layer + grad check and updated 2NN example with nesterov sgd.

### CODE REVIEW

**Comment by j143** [comment]
Hi @tobiasrieger - Thanks a lot for contributing deep learning examples.

**TL;DR:** Providing more info would be easier to review the changes! 😄 

1. [New feature guideline](https://github.com/apache/systemds/blob/master/CONTRIBUTING.md#general-guidelines-and-philosophy-for-contribution)

"New features (e.g., **a new cutting edge machine learning algorithm**) typically will live in scripts/staging or its equivalent folder for specific feature to get **some airtime** and **sufficient testing** before a decision is made regarding whether they are to migrated to the top-level."

2. The end goal is to make sure the contributions will add a base for further research.
For this reason, try adding any
    - reference papers,
    - docs,
    - articles
    - screenshots
    - rationale

An example [documentation](https://github.com/apache/systemds/blob/master/src/main/python/docs/source/onnx_systemds/onnx_systemds_design.rst#design).

Thank you.

File: scripts/nn/examples/Example-MNIST_2NN_Leaky_ReLu_Softmax.dml [hunk 1]
```diff
@@ -20,12 +20,15 @@
...
 
 /*
- * The MNIST Data can be downloaded as follows:
- * mkdir -p data/mnist/
- * cd data/mnist/
- * curl -O https://pjreddie.com/media/files/mnist_train.csv
- * curl -O https://pjreddie.com/media/files/mnist_test.csv
- */
+* This Example trains a feed forward neural network with one input layer, two hidden affine layers (200 neurons) with
+* leaky relu activations and one affine output layer with a softmax activation
```
Comment (j143) [inline scripts/nn/examples/Example-MNIST_2NN_Leaky_ReLu_Softmax.dml:24]: These changes are only telling us what is added.

But, not WHY these changes are added. If you have contacted any SystemDS developer offline, consider providing
the summary of the discussion here.

File: scripts/nn/examples/mnist_2NN.dml [hunk 2]
```diff
@@ -20,159 +20,161 @@
...
+*  - Y: Target matrix, of shape (N, K).
+*  - X_val: Input validation data matrix, of shape (N, C*Hin*Win).
+*  - Y_val: Target validation matrix, of shape (N, K).
+*  - epochs: Total number of full training loops over the full data set.
+*
+* Outputs:
+*  - W: Weights (parameters) matrix, of shape (D, M, 3).
+*  - b: Biases vector, of shape (1, M, 3).
+*/
+
```
Comment (j143) [inline scripts/nn/examples/mnist_2NN.dml:57]: This seems the formatting change. Can you revert this?

**Review by j143 — COMMENTED** [review]
Hi @tobiasrieger - this is a cursory review.

1. Let us follow the existing formatting of the code, since most of our code is in the same.
2. Thank you for adding the gradient check. :smile:
3. https://github.com/apache/systemds/pull/1026#issuecomment-675005848

**Comment by mboehm7** [comment]
LGTM - thanks for the extensions @tobiasrieger. I only changed the formatting of comments, and simplified `ifelse(X > 0, X, 0.01 * X)` to `max(X, 0.01 * X)`.
